pkgs/by-name/sp/spliit/package.nix

```diff
@@ -0,0 +1,83 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+  openssl,
+  prisma,
+  prisma-engines,
+  writeShellScriptBin,
+}:
+
+buildNpmPackage rec {
+  pname = "spliit";
+  version = "1.15.0";
+
+  src = fetchFromGitHub {
+    owner = "spliit-app";
+    repo = "spliit";
+    tag = version;
+    hash = "sha256-rFnaYmS0IA2Bh6+aOUVsJRYlBHzMPDTJzdCyPhBD8LA=";
+  };
+
+  preBuild = ''
+    prisma generate
+    # The build.env file is required for the building, during runtime it will not be used.
+    cp ./scripts/build.env .env
+  '';
+
+  npmDepsHash = "sha256-0wMd6bxWsIv90eHVt95Y8//AUERx2P9LpzeO05DIm1U=";
+
+  buildInputs = [
+    openssl
+  ];
+
+  nativeBuildInputs = [
+    prisma
+  ];
+
+  env = {
+    NEXT_TELEMETRY_DISABLED = 1;
+  };
+
+  installPhase =
+    let
+      startScript = writeShellScriptBin "spliit-start-script" ''
+        set -eu pipefail
+
+        export PRISMA_SCHEMA_ENGINE_BINARY="${prisma-engines}/bin/schema-engine";
+        export PRISMA_QUERY_ENGINE_BINARY="${prisma-engines}/bin/query-engine";
+        export PRISMA_QUERY_ENGINE_LIBRARY="${prisma-engines}/lib/libquery_engine.node";
+        export PRISMA_FMT_BINARY="${prisma-engines}/bin/prisma-fmt";
+
+        cd "$(dirname "$0")/.."
```

Comment:
What about using `makeWrapper` with `--set` and `--chdir` (unsure)?

Reply:
Works quite well if you add the flags in the correct order.